### DISCUSSION TITLE
enctrypted stratis + swapfile on root + dracut resume crashes boot

### DISCUSSION THREAD

**Discussion by fneddy** [stratis-storage/project#432] [Q&A]
if all of this happens then bootig is not longer possible:

1. root is on stratis
2. stratis pool is encrypted
3. swap is on a swapfile on root fileystem
4. **stratis** and **resume** modules are added to dracut
5. kernel parameters **resume** and **resume_offset** are set

if all of this happens its not possible to boot the system any more. i dont get the prompt for entering the decryption password. even when **rhgb** and **quiet** parameters are not set. 

if i add the dracut debug flag **rd.debug** i see there is a loop running at 0.5 seconds that tries to dercrypt root. but its not possible to interact anymore.

it looks like the dracut password promt somehow does not block and wait for the password and the resume module already tries to do things on a filesystem that is not decrypted.

**Comment by jbaublitz** [comment]
Hi @fneddy! I am mainly wondering right off the bat if this is a dracut ordering issue. I'd also be curious if resume_offset requires the pool to be unencrypted. If it does not, it is likely going through crypttab which we do not support due to the format not being able to represent our many device to one pool relationship. Can you give us additional logs and any other information you might deem relevant from `rd.debug`?

**Comment by fneddy** [comment]
Hi, i did some more testing. i seems that the dracut resume module is broken. i tried a setup with luks2+ext4 and swapfile. this sucessfully asks for a password but then fails to boot after the root was decrypted. i will open a ticket at dracut repo and when this is fixed i'll report back here.

**Reply by jbaublitz** [reply]
Thanks for letting us know and please keep us posted!

**Comment by fneddy** [comment]
ok, i just realized that there is already a ticket at [dracut repo](https://github.com/dracutdevs/dracut/issues/924) about swapfile on encrypted root.
